### PR TITLE
fix: [PL-26472]: Removing appRole Token renewal

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35507)

### CODE REVIEW

**Review by gn-harness — CHANGES_REQUESTED** [review]
> Reviewed 3 of 16 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35507)* status: 16 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java` line 172 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LOBBNBbiPJv7BA9C9:-N8LOBBNBbiPJv7BA9CA:b-bdyay2) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java#L172)):*
> ```Java
>       ConnectorDTO connectorDTO, String accountIdentifier, ChangeType gitChangeType) {
>     if (ngFeatureFlagHelperService.isEnabled(accountIdentifier, REMOVE_APPROLE_TOKEN_RENEWAL)
>         && (connectorDTO.getConnectorInfo().getConnectorConfig() instanceof VaultConnectorDTO)) {
> ```

unit test?



_Code quote:_
```Java
  private ConnectorResponseDTO createInternal(
      ConnectorDTO connectorDTO, String accountIdentifier, ChangeType gitChangeType) {
    if (ngFeatureFlagHelperService.isEnabled(accountIdentifier, REMOVE_APPROLE_TOKEN_RENEWAL)
        && (connectorDTO.getConnectorInfo().getConnectorConfig() instanceof VaultConnectorDTO)) {
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java` line 180 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LFvuECdxseR5iSyUH:-N8LFvuECdxseR5iSyUI:bcjlc8j) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java#L180)):*
> ```Java
>         connectorDTO.setConnectorInfo(connectorInfoDTO);
>       }
>     }
> ```

nit: This can be carved out in its own helper function as your sharing the same code with update method. something like skipAppRoleRenewalForVaultConnector



_Code quote:_
```Java
    if (ngFeatureFlagHelperService.isEnabled(accountIdentifier, REMOVE_APPROLE_TOKEN_RENEWAL)
        && (connectorDTO.getConnectorInfo().getConnectorConfig() instanceof VaultConnectorDTO)) {
      ConnectorInfoDTO connectorInfoDTO = connectorDTO.getConnectorInfo();
      VaultConnectorDTO vaultConnectorDTO = (VaultConnectorDTO) connectorInfoDTO.getConnectorConfig();
      if (AccessType.APP_ROLE.equals(vaultConnectorDTO.getAccessType())) {
        vaultConnectorDTO.setDoNotRenewAppRoleToken(true);
        connectorInfoDTO.setConnectorConfig(vaultConnectorDTO);
        connectorDTO.setConnectorInfo(connectorInfoDTO);
      }
    }
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java` line 90 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LHPno2TxFEqqgr5QJ:-N8LHPno2TxFEqqgr5QK:b3at9vi) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java#L90)):*
> ```Java
>           "Vault {} configured with Vault-Agent or Aws Iam Auth. It does not need renewal", vaultConnector.getUuid());
>     } else {
>       vaultConnector = mongoTemplate.findById(vaultConnector.getId(), VaultConnector.class);
> ```

probably unrelated to your change -- but the else block seems redundant. Unnecessary nested makes it harder to read and maintain.

if (isRenewalNotNeeded) {
  return
}
...



_Code quote:_
```Java
    } else {
      vaultConnector = mongoTemplate.findById(vaultConnector.getId(), VaultConnector.class);
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java` line 104 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LNUzpDrJpKuZAblSQ:-N8LNUzpDrJpKuZAblSR:bbydo9t) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java#L104)):*
> ```Java
>         }
>         if (vaultConnector.getAccessType() == APP_ROLE) {
>           if (!vaultConnector.getDoNotRenewAppRoleToken()) {
> ```

Would there be any path where accessType would APP_ROLE and doNotRenewAppRoleToken is set to false for VaultConnector? Just making sure we are additional config because of a valid usecase



_Code quote:_
```Java
        if (vaultConnector.getAccessType() == APP_ROLE) {
          if (!vaultConnector.getDoNotRenewAppRoleToken()) {
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java` line 104 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LOEOo4uc4vwRQudUh:-N8LOEOo4uc4vwRQudUi:b-bdyay2) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java#L104)):*
> ```Java
>         }
>         if (vaultConnector.getAccessType() == APP_ROLE) {
>           if (!vaultConnector.getDoNotRenewAppRoleToken()) {
> ```

unit test?



_Code quote:_
```Java
          if (!vaultConnector.getDoNotRenewAppRoleToken()) {
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LH2BVCazby_uplhck:-N8LH2BVCazby_uplhcl:b3s14oo) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java#L106)):*
> ```Java
>           if (!vaultConnector.getDoNotRenewAppRoleToken()) {
>             vaultService.renewAppRoleClientToken(vaultConnector);
>           }
> ```

This can be clubbed with the isRenewalNotNeeded to return early.



_Code quote:_
```Java
          if (!vaultConnector.getDoNotRenewAppRoleToken()) {
            vaultService.renewAppRoleClientToken(vaultConnector);
          }
```
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/vaultconnector/VaultConnector.java` line 134 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LHiDxCuL7b7uJv4ds:-N8LHiDxCuL7b7uJv4dt:bnkx8s8) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/vaultconnector/VaultConnector.java#L134)):*
> ```Java
> 
>   public Boolean getDoNotRenewAppRoleToken() {
>     return doNotRenewAppRoleToken == null ? Boolean.FALSE : doNotRenewAppRoleToken;
> ```

Can this be null if you have a default set?



_Code quote:_
```Java
doNotRenewAppRoleToken == null ? 
```

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35507)* status: 16 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java` line 90 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LHPno2TxFEqqgr5QJ:-N8Ms1G7AKBeSFtwM5Ow:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java#L90)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

probably unrelated to your change -- but the else block seems redundant. Unnecessary nested makes it harder to read and maintain.

if (isRenewalNotNeeded) {
  return
}
...
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LH2BVCazby_uplhck:-N8MtFuiEAqUazT1Bry1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java#L106)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

This can be clubbed with the isRenewalNotNeeded to return early.
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/vaultconnector/VaultConnector.java` line 134 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LHiDxCuL7b7uJv4ds:-N8Mtmsk2BbjpojkWbXY:bcc1wn2) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/vaultconnector/VaultConnector.java#L134)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

Can this be null if you have a default set?
</blockquote></details>

It can be null for the existing vault entries in dB. For those entities we will be setting this value as false.

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35507)

**Review by gn-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35507)* status: 18 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java` line 104 at r3](https://reviewable.io/reviews/harness/harness-core/35507#-N8LNUzpDrJpKuZAblSQ:-N8Pxjp_1eeEAseyrQ_U:b499ea6) ([raw file](https://github.com/harness/harness-core/blob/792544f9e65449e2a834af8ce305cfc8076c6987/120-ng-manager/src/main/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandler.java#L104)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

Would there be any path where accessType would APP_ROLE and doNotRenewAppRoleToken is set to false for VaultConnector? Just making sure we are additional config because of a valid usecase
</blockquote></details>

Just to clarify this one -- do you need doNotRenewAppRoleToken as a config? @nishantsaini20 thoughts?

This is the only blocking change, everything else looks good.
___
*[`120-ng-manager/src/test/java/io/harness/ng/ConnectorServiceImplTest.java` line 65 at r6](https://reviewable.io/reviews/harness/harness-core/35507#-N8Q-2MpD_lfilYMfwhO:-N8Q-2MpD_lfilYMfwhP:b-ya68c3) ([raw file](https://github.com/harness/harness-core/blob/f90bef8561ccc0b3cc6c27a157d5f9f4c8b3fe4d/120-ng-manager/src/test/java/io/harness/ng/ConnectorServiceImplTest.java#L65)):*
> ```Java
>         connectorActivityService, connectorHeartbeatService, connectorRepository, eventProducer, executorService,
>         connectorErrorMessagesHelper, harnessManagedConnectorHelper, ngErrorHelper, gitSyncSdkService,
>         instrumentationHelper, opaConnectorService, ngFeatureFlagHelperService));
> ```

Does InjectMocks not suffice?



_Code quote:_
```Java
    connectorService = spy(new ConnectorServiceImpl(defaultConnectorService, secretManagerConnectorService,
        connectorActivityService, connectorHeartbeatService, connectorRepository, eventProducer, executorService,
        connectorErrorMessagesHelper, harnessManagedConnectorHelper, ngErrorHelper, gitSyncSdkService,
        instrumentationHelper, opaConnectorService, ngFeatureFlagHelperService));
```
___
*[`120-ng-manager/src/test/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandlerTest.java` line 108 at r6](https://reviewable.io/reviews/harness/harness-core/35507#-N8Q3U244Xa3qSAb8I0a:-N8Q3U244Xa3qSAb8I0b:b-fwfp8u) ([raw file](https://github.com/harness/harness-core/blob/f90bef8561ccc0b3cc6c27a157d5f9f4c8b3fe4d/120-ng-manager/src/test/java/io/harness/ng/core/handler/NGVaultSecretManagerRenewalHandlerTest.java#L108)):*
> ```Java
>   @Category(UnitTests.class)
>   public void testRenewalForAppRole_doNotRenew_doNotRenewAppRoleSetToTrue() {
>     VaultConnector vaultConnector = getAppRoleBasedVault();
> ```

nit:

getDefaultAppRoleBasedVault().toBuilder().doNotRenewAppRoleToken(false).build();

Basically, your test setup should be inside the test and one should not have to read another function to see the relevant details.



_Code quote:_
```Java
getAppRoleBasedVault()
```

**Comment by VikasMaddukuriHarness** [comment]
trigger authorcheck
